runners/spark/src/test/java/org/apache/beam/runners/spark/ProvidedSparkContextTest.java

```diff
@@ -64,21 +47,25 @@ private SparkContextOptions getSparkContextOptions(JavaSparkContext jsc) {
     @Test
     public void testWithProvidedContext() throws Exception {
         JavaSparkContext jsc = new JavaSparkContext("local[*]", "Existing_Context");
+        testWithValidProvidedContext(jsc);
+        if (!jsc.sc().isStopped()) {
```

Comment:
Asserting here that the context is not stopped would be good, and make `testReuseProvidedContext` redundant I believe - you want to make sure that context remains alive after executing a pipeline with a provided context & that also guarantees it's reusable..